### PR TITLE
a fix for undo issue

### PR DESCRIPTION
**Pull request by Ariex**
Hi, this is a fix for the undo issue described in https://github.com/dsandstrom/atom-double-tag/issues/6

currently I set the transact to 1 second, so if you modify tag a little bit (opt1), then wait for 1 sec, then do another modify (opt2), you can undo them as two operations rather than letter by letter.

tested locally, all works fine, 
## known bug
- there is a "empty" operation if you undo mutiple operations, so you can "ctrl+z" undo opt2, then next "ctrl-z" is empty, then next "ctrl+z" will undo opt1 for you

### CODE REVIEW

**Comment by dsandstrom** [comment]
Thanks, I'll test it tomorrow.

File: lib/double-tag.coffee [hunk 1]
```diff
@@ -18,7 +18,6 @@ class DoubleTag
     @subscriptions.add @editor.onDidChangeCursorPosition (event) =>
       @reset() if @foundTag and @cursorLeftMarker()
       return if @foundTag
-
```
Comment (dsandstrom) [inline lib/double-tag.coffee]: Why are the lines removed?

File: spec/double-tag-spec.coffee [hunk 2]
```diff
@@ -212,3 +212,17 @@ describe "DoubleTag", ->
...
+    describe "undo-test", ->
+      beforeEach ->
+        editor.setText('<div>test</div>')
+
+      describe "when undo", ->
+        it "should undo in block", ->
+          editor.setCursorBufferPosition([0, 4])
+          editor.insertText('-span')
+          editor.undo()
+          # not sure why need to undo here, but in editor, one "ctrl+z" is good enough
```
Comment (dsandstrom) [inline spec/double-tag-spec.coffee]: Maybe because of the 1000ms delay?  Is there a way to force the undo right away in tests?

**Comment by dsandstrom** [comment]
Working better so far

File: lib/double-tag.coffee (hunk 1 above)
Comment (Ariex) [inline lib/double-tag.coffee]: I think this is an empty line, maybe my atom-beauty removed it...

File: spec/double-tag-spec.coffee (hunk 2 above)
Comment (Ariex) [inline spec/double-tag-spec.coffee]: 1000ms delay is only for the transact, I have tested when type thing quickly then press 'ctrl+z', it all works fine...
from the output, the first undo seems undo the change of the closing tag, the 2nd one undo the change in begnning tag.

**Comment by Ariex** [comment]
you could view this pr as a example to solving the issue since it does not solve all issues, hope this can give you an idea to solve this issue in a perfect way; also a pr in master may be wrong...
i think for the double undo issue, it might be, the first undo reset changes in end tag, and this extension updated the beginning tag, so it looks like one undo did all the job and second undo did nothing

**Comment by dsandstrom** [comment]
Thanks for the help.  Yes, you should create a new branch when making a PR.  It's more for your benefit, than mine.  Your master is ahead of my master so you would have re-fork this repo in order to make another PR.

I will use this as a starting off point in order to improve undo.

**Comment by mrleblanc101** [comment]
Any news on this fix ?

**Comment by dsandstrom** [comment]
Fixed in another way.
